### ISSUE TITLE
Library window position state isn't saved/restored

### ISSUE THREAD

**Issue by cassidyjames** [elementary/photos#566]
It looks like we save the size of the library window, but not the position.

**Comment by 4jNsY6fCVqZv** [comment]
Hello, I'd like to work on a solution. Probably the appropriate place is https://github.com/elementary/photos/blob/master/data/io.elementary.photos.gschema.xml, isn't it?
Is there some kind of a code reference in other elementary apps that helps me to make the necessary changes?

**Comment by cassidyjames** [comment]
Yeah, here is the GSchema where we are saving the size of the library window:

https://github.com/elementary/photos/blob/c6b0381baf79e62d1d7cd4bbfd4502da0ecc365f/data/io.elementary.photos.gschema.xml#L155-L165

Here is a recent pull request where someone updated how Code saves its window state: https://github.com/elementary/code/pull/794

It looks like you'd want to update LibraryWindow.vala here:

https://github.com/elementary/photos/blob/b62cc8f0119490ce1f959abc3cbe43be7748e24c/src/library/LibraryWindow.vala#L146-L149

and here:

https://github.com/elementary/photos/blob/b62cc8f0119490ce1f959abc3cbe43be7748e24c/src/library/LibraryWindow.vala#L514-L520

…to save and restore the window position in addition to the width/height.

**Comment by 4jNsY6fCVqZv** [comment]
@cassidyjames Should the key name ideally stay with "library-xxx"? I have seen that other apps tend to use something like "window-xxx" instead.

**Comment by cassidyjames** [comment]
@4jNsY6fCVqZv for Photos it looks like this state is specifically for the Library window, i.e. not the previewer window. So the current name makes sense to me. Default sizes are up to the different apps and their own use cases and designs—I don't think it's necessary to have them all be the same default size.

**Comment by 4jNsY6fCVqZv** [comment]
@cassidyjames 
> Default sizes are up to the different apps and their own use cases and designs

The differences are so marginal, does that really have something to do with design and use cases? This is not self-explanatory for me yet.

**Comment by 4jNsY6fCVqZv** [comment]
@cassidyjames Question for better understanding, after a look into different elementary default apps has confused me a bit, because the solutions sometimes look very different: The current LibraryWindow.vala solution already saves/restores the status of the window size and it's just a matter of adding the position value, right? Or is it also a matter of introducing saving/restoring as such?

**Comment by cassidyjames** [comment]
>The current LibraryWindow.vala solution already saves/restores the status of the window size and it's just a matter of adding the position value, right?

@4jNsY6fCVqZv that's correct!

**Comment by 4jNsY6fCVqZv** [comment]
@cassidyjames 
When I look at https://elementary.io/docs/code/reference, my solution still feels a bit fragmentary to me. Could you imagine taking a look at my first draft? https://github.com/elementary/photos/pull/569

**Comment by danirabbit** [comment]
Closing since windows can't position themselves in Wayland so these days Gala saves window positions
